sdk/test/trace/tracer_test.cc

```diff
@@ -519,6 +519,396 @@ TEST(Tracer, SpanSetLinks)
+      auto attrs = link.GetAttributes();
+      ASSERT_EQ(attrs.size(), 1);
+      ASSERT_EQ(nostd::get<int>(attrs.at("attr11")), 11);
+    }
+
+    {
+      auto link  = span_data_links.at(2);
+      auto attrs = link.GetAttributes();
+      ASSERT_EQ(attrs.size(), 1);
+      ASSERT_EQ(nostd::get<int>(attrs.at("attr12")), 12);
+    }
+
+    {
+      auto link  = span_data_links.at(3);
+      auto attrs = link.GetAttributes();
+      ASSERT_EQ(attrs.size(), 1);
+      ASSERT_EQ(nostd::get<int>(attrs.at("attr13")), 13);
+    }
+
+    {
+      auto link  = span_data_links.at(4);
+      auto attrs = link.GetAttributes();
+      ASSERT_EQ(attrs.size(), 1);
+      ASSERT_EQ(nostd::get<int>(attrs.at("attr14")), 14);
+    }
+
+    {
+      auto link  = span_data_links.at(5);
+      auto attrs = link.GetAttributes();
+      ASSERT_EQ(attrs.size(), 1);
+      ASSERT_EQ(nostd::get<int>(attrs.at("attr15")), 15);
+    }
+
+    {
+      auto link  = span_data_links.at(6);
+      auto attrs = link.GetAttributes();
+      ASSERT_EQ(attrs.size(), 2);
+      ASSERT_EQ(nostd::get<int>(attrs.at("attr16")), 16);
+      ASSERT_EQ(nostd::get<int>(attrs.at("attr17")), 17);
+    }
+
+    {
+      auto link  = span_data_links.at(7);
+      auto attrs = link.GetAttributes();
+      ASSERT_EQ(attrs.size(), 1);
+      ASSERT_EQ(nostd::get<int>(attrs.at("attr18")), 18);
+    }
+
+    {
+      auto link  = span_data_links.at(8);
+      auto attrs = link.GetAttributes();
+      ASSERT_EQ(attrs.size(), 2);
+      ASSERT_EQ(nostd::get<int>(attrs.at("attr19")), 19);
+      ASSERT_EQ(nostd::get<int>(attrs.at("attr20")), 20);
+    }
+
+    {
+      auto link  = span_data_links.at(9);
+      auto attrs = link.GetAttributes();
+      ASSERT_EQ(attrs.size(), 1);
+      ASSERT_EQ(nostd::get<int>(attrs.at("attr21")), 21);
+    }
+
+    {
+      auto link  = span_data_links.at(10);
+      auto attrs = link.GetAttributes();
+      ASSERT_EQ(attrs.size(), 2);
+      ASSERT_EQ(nostd::get<std::string>(attrs.at("attra1")), "1");
+      ASSERT_EQ(nostd::get<std::string>(attrs.at("attra2")), "2");
+    }
+
+    {
+      auto link  = span_data_links.at(11);
+      auto attrs = link.GetAttributes();
+      ASSERT_EQ(attrs.size(), 2);
+      ASSERT_EQ(nostd::get<std::string>(attrs.at("attra3")), "3");
+      ASSERT_EQ(nostd::get<std::string>(attrs.at("attra4")), "4");
+    }
+
+    {
+      auto link  = span_data_links.at(12);
+      auto attrs = link.GetAttributes();
+      ASSERT_EQ(attrs.size(), 2);
+      ASSERT_EQ(nostd::get<std::string>(attrs.at("attrb1")), "1");
+      ASSERT_EQ(nostd::get<std::string>(attrs.at("attrb2")), "2");
+    }
+
+    {
+      auto link  = span_data_links.at(13);
+      auto attrs = link.GetAttributes();
+      ASSERT_EQ(attrs.size(), 2);
+      ASSERT_EQ(nostd::get<std::string>(attrs.at("attrb3")), "3");
+      ASSERT_EQ(nostd::get<std::string>(attrs.at("attrb4")), "4");
+    }
+  }
+}
+#endif /* OPENTELEMETRY_ABI_VERSION_NO >= 2 */
```

Comment:
nit, this #endif is not needed and can be merged with the following one.

Reply:
Fixed.